core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -98,87 +159,243 @@ public GeoBoundingBoxQueryBuilder bottomRight(String geohash) {
-        // check values
-        if(Double.isNaN(box[TOP])) {
-            throw new IllegalArgumentException("geo_bounding_box requires top latitude to be set");
-        } else if(Double.isNaN(box[BOTTOM])) {
-            throw new IllegalArgumentException("geo_bounding_box requires bottom latitude to be set");
-        } else if(Double.isNaN(box[RIGHT])) {
-            throw new IllegalArgumentException("geo_bounding_box requires right longitude to be set");
-        } else if(Double.isNaN(box[LEFT])) {
-            throw new IllegalArgumentException("geo_bounding_box requires left longitude to be set");
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        // validation was not available prior to 2.x, so to support bwc percolation queries we only ignore_malformed on 2.x created indexes
+        if (!ignoreMalformed) {
+            if (Numbers.isValidDouble(topLeft.getLat()) == false) {
+                validationException = addValidationError("top latitude is invalid number: " + topLeft.getLat(), validationException);
+            }
+            if (Numbers.isValidDouble(topLeft.getLon()) == false) {
+                validationException = addValidationError("left longitude is invalid number: " + topLeft.getLon(), validationException);
+            }
+            if (Numbers.isValidDouble(bottomRight.getLat()) == false) {
+                validationException = addValidationError("bottom latitude is invalid number: " + bottomRight.getLat(), validationException);
+            }
+            if (Numbers.isValidDouble(bottomRight.getLon()) == false) {
+                validationException = addValidationError("right longitude is invalid number: " + bottomRight.getLon(), validationException);
+            }
+
+            if (validationException == null) {
+                // all corners are valid after above checks - make sure they are in the right relation
+                if (topLeft.getLat() < bottomRight.getLat()) {
+                    validationException = addValidationError("top is below bottom corner: " +
+                            topLeft.getLat() + " vs. " + bottomRight.getLat(), validationException);
+                }
+
+                // we do not check longitudes as the query generation code can deal with flipped left/right values
+            }
         }
 
-        builder.startObject(NAME);
+        if (Strings.isEmpty(fieldName)) {
+            validationException = addValidationError("field name must be non-null and non-empty", validationException);
+        }
+        return validationException;
+    }
 
-        builder.startObject(name);
-        builder.array(TOP_LEFT, box[LEFT], box[TOP]);
-        builder.array(BOTTOM_RIGHT, box[RIGHT], box[BOTTOM]);
-        builder.endObject();
+    QueryValidationException validate(boolean indexCreatedBeforeV2_0) {
+        if (ignoreMalformed || indexCreatedBeforeV2_0) {
```

Comment:
Ahh I see what you've done now sorry, I hadn't seen that this is called from the toQuery() method below. I think the method name should be changed though as given that it has the same name as the above validate() method it is confusing as to when it would be called